packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -5852,31 +5852,34 @@ export class ThinEngine {
     }
 
     /**
-     * Queue a new function into the requested animation frame pool (ie. this function will be executed byt the browser for the next frame)
+     * Queue a new function into the requested animation frame pool (ie. this function will be executed by the browser (or the javascript engine) for the next frame)
      * @param func - the function to be called
      * @param requester - the object that will request the next frame. Falls back to window.
      * @returns frame number
      */
     public static QueueNewFrame(func: () => void, requester?: any): number {
+        // Note that there is kind of a typing issue here, as `setTimeout` might return something else than a number (NodeJs returns a NodeJS.Timeout object).
+        // Also if the global `requestAnimationFrame`'s returnType is number, `requester.requestPostAnimationFrame` and `requester.requestAnimationFrame` types
+        // are `any`.
+
         if (!IsWindowObjectExist()) {
-            if (typeof requestAnimationFrame !== "undefined") {
+            if (typeof requestAnimationFrame === "function") {
                 return requestAnimationFrame(func);
             }
-
-            return setTimeout(func, 16) as unknown as number;
         }
-
-        if (!requester) {
-            requester = window;
+        else {
+            const { requestPostAnimationFrame, requestAnimationFrame } = requester || window;
+            if (typeof requestPostAnimationFrame === 'function') {
+                return requestPostAnimationFrame(func);
+            }
+            if (typeof requestAnimationFrame === 'function') {
+                return requestAnimationFrame(func);
+            }
         }
 
-        if (requester.requestPostAnimationFrame) {
-            return requester.requestPostAnimationFrame(func);
-        } else if (requester.requestAnimationFrame) {
-            return requester.requestAnimationFrame(func);
-        } else {
-            return window.setTimeout(func, 16);
-        }
+        // fallback to the global `setTimeout`.
+        // In most cases (aka in the browser), `window` is the global object, so instead of calling `window.setTimeout` we could call the global `setTimeout`.
+        return setTimeout(func, 16) as unknown as number;

```

Comment:
Here are the main changes of this PR.

We now test that the properties are of type `function` before calling them, which is safer but might be overkill.
And we do not call `window.setTimeout` anymore, we directly fallback to the global `setTimeout`.

If that's not ok, before the fallback I can add something like
```
if (typeof window.setTimeout === 'function') {
    return window.setTimeout(func);
}
```